docs: Adds docs for text field suffix (#381)
holaluz/margarita: docs/components/MaTextFieldWithSuffix.docs.mdx

> # TextField-with-suffix

Reviewer: ```suggestion
# TextField with a suffix
```

## Description

> Input field with independent (from the value) and fixed (always visible) suffix. 

Reviewer: Not sure this sentence makes sense 🤔  I would change it to something like:
`Input field with a suffix section which is always displayed/visible `

Reviewer 2: I'd rather say `Input field with a suffix` and that's all. It is visible as it might be expected, isn't it?

Reviewer: You're right, straight to the point, there is no need in saying so 💯

> The suffix itself can be either text, symbols or both.

Reviewer: ```suggestion
The suffix property accepts both text and symbols.
```

## How to use it? (Controls)

We can edit the same items from the text-field + the **suffix**.

- Aria label
- Edit label
- Edit placeholder value
- Add and edit suffix
- Error message
- State

> If the user misses that the input already has a suffix and enters it → Allow this in our validation and do not show an error.

Reviewer: Shall we add this validation into our component in an incoming iteration?

⚠️  **How not to use it**

- Do not rely only on suffixes, some users will not read them.

> - Suffixes will not replace labels. If you need a specific type of information, say so in the input label or hint/helper text as well.

Reviewer: ```suggestion
- Suffix property does not replace the label's responsibilities. If you need a specific type of data, empathise it in the input's label or placeholder.
```

## When to use it

- When there is a commonly understood symbol or abbreviation for the type of information we are asking for.

> - When the user needs to enter a value which has specific units.

Reviewer: ```suggestion
- When the user needs to enter a value that has a specific unit.
```

***For example:***

The user wants to put solar panels on their rooftop and the total cost is 8.000€. They need to specify how much they want to pay at the moment (the rest will be financed).

👉 The user needs to enter an amount and whatever the amount → it will be € → so we fill the unit for them.

## Variants

[https://www.figma.com/embed?embed_host=notion&url=https%3A%2F%2Fwww.figma.com%2Ffile%2F7RhweKibaUnxX6WYBl7zKW%2FDesign-System-Beta%3Fnode-id%3D2572%253A1707](https://www.figma.com/embed?embed_host=notion&url=https%3A%2F%2Fwww.figma.com%2Ffile%2F7RhweKibaUnxX6WYBl7zKW%2FDesign-System-Beta%3Fnode-id%3D2572%253A1707)

## Behavior

> Same as text-field.

Reviewer: Could we add the link to our `TextField` documentation?

> ## References

Reviewer: Not sure this is needed in our components library documentation. However, I believe it would be nice to have it referenced in our Figma design.

Reviewer 2: This documentation was written by the Design team so I understand that they want this to be part of the docs too.

Reviewer: If they think that this should be here, fine by me! 🚀  I just found that a bit odd.
